ksqldb-engine/src/test/java/io/confluent/ksql/integration/JoinIntTest.java

```diff
@@ -82,8 +85,10 @@ public void before() {
     TEST_HARNESS.ensureTopics(itemTableTopicJson, itemTableTopicAvro,
         orderStreamTopicJson, orderStreamTopicAvro);
 
-    TEST_HARNESS.produceRows(itemTableTopicJson, ITEM_DATA_PROVIDER, JSON, () -> now - 500);
-    TEST_HARNESS.produceRows(itemTableTopicAvro, ITEM_DATA_PROVIDER, AVRO, () -> now - 500);
+    // we want the table events to always be present (less than the ts in the stream
+    // including the time extractor)
+    TEST_HARNESS.produceRows(itemTableTopicJson, ITEM_DATA_PROVIDER, JSON, () -> 0L);
```

Comment:
Judging by your description, any stream event with timestamp greater than 0 shouldn't be able to join with this, right?

Are you assuming that once you address the other bug, this should fail?  (That would validate the assertion that the join shouldn't happen with this timestamp)

Reply:
> Judging by your description, any stream event with timestamp greater than 0 shouldn't be able to join with this, right?

I'm not sure I understand - it would logically be the opposite? If the timestamp of the table event is less than the stream event timestamp then it should be able to see the table entry.

> Are you assuming that once you address the other bug, this should fail? (That would validate the assertion that the join shouldn't happen with this timestamp)

What other bug? I'm not sure there is a bug, something changed in behavior and I'm trying to understand if what changed is a bug or not.

Reply:
You're right, I got that backward.

The other bug I was referring to was this:
> The weird part, is that it works if we set the timestamp to anything less than 9L - meaning that if my stream event has timestamp 1L it can join with a table entry that has timestamp 8L only if there exists an event in the stream that has a timestamp greater than or equal to 8L.

I was mainly pointing out that if this was a bug, it might make sense to set a value that should not work (but does!), so that if this is uncovered in the future, the test will break.  Or maybe that should be its own test.

Just a thought.

Reply:
> I was mainly pointing out that if this was a bug, it might make sense to set a value that should not work (but does!), so that if this is uncovered in the future, the test will break. Or maybe that should be its own test.

ah, got it! yeah that would be a valuable test to add once we understand what's going on.